### PR TITLE
Expand GetUser and GetUserScores requests by adding RulesetInfo parameter

### PR DESCRIPTION
**Pull request by EVAST9919**
Split from https://github.com/ppy/osu/pull/5238

### CODE REVIEW

File: osu.Game/Online/API/Requests/GetUserRequest.cs [hunk 1]
```diff
@@ -2,18 +2,21 @@
...
 
-        public GetUserRequest(long? userId = null)
+        public GetUserRequest(long? userId = null, RulesetInfo ruleset = null)
         {
             this.userId = userId;
+            this.ruleset = ruleset;
         }
 
-        protected override string Target => userId.HasValue ? $@"users/{userId}" : @"me";
+        protected override string Target => userId.HasValue ? (ruleset != null ? $@"users/{userId}/{ruleset.ShortName}" : $@"users/{userId}") : @"me";
```
Comment (iiSaLMaN) [inline osu.Game/Online/API/Requests/GetUserRequest.cs]: What if we requested a null user id with non-null ruleset? isn't that what we should do when retrieving info of `me` with a specific ruleset?

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/Online/API/Requests/GetUserRequest.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Online/API/Requests/GetUserRequest.cs]: Can this not just be `users/{(userId ?? "me")}/{ruleset?.ShortName}`

**Comment by peppy** [comment]
test?

**Comment by peppy** [comment]
as in run your code before you push? please?

**Comment by EVAST9919** [comment]
> as in run your code before you push? please?

I do and it works fine for me

**Review by peppy — DISMISSED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**